### PR TITLE
UPC-134 Add Scanner-Languages manifest property

### PR DESCRIPTION
**Pull request by cody-simms-sonarsource**

### CODE REVIEW

File: sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/PluginManifest.java [hunk 1]
```diff
@@ -417,6 +426,22 @@ public PluginManifest setDevelopers(String[] developers) {
     return this;
   }
 
+  /**
+   * @since 1.31
+   */
+  public String[] getLanguages() {
```
Comment (dbmeneses) [inline sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/PluginManifest.java:432]: I think that the name `languages` is not very clear (difficult to understand what it means).
Even the purpose of the field itself, `Scanner-Languages` isn't obvious (and I know I was the one proposing it!).

Any thoughts on that?

Comment (dbmeneses) [reply sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/PluginManifest.java:432]: naming is difficult ...

Comment (cody-simms-sonarsource) [reply sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/PluginManifest.java:432]: I came up with some ideas, let me know what you think:
* `applicableLanguages` - In other words, languages that are pertinent to the plugin
* `requiredForLanguages` - Trying to convey the fact that this plugin should always be downloaded for the specified languages
* `requiredByLanguages` - Similar to the last one, but kind of flipped - "These are the languages that if detected, will cause the scanner engine to download this plugin"

I think I like the last one the best.

As for the manifest property name, I'll change it to match whatever the PluginManifest field name we decide on - but I noticed that almost all the other property names start with "Plugin-", is there a reason we don't want this one to as well?

Comment (dbmeneses) [reply sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/PluginManifest.java:432]: I like `requiredForLanguages`!
And I hadn't noticed that most manifest property names start with "Plugin-" - good idea to follow that convention to keep it consistent.

**Review by dbmeneses — APPROVED** [review]

**Pull request merged**